test/functional/p2p_filter.py

```diff
@@ -64,19 +64,40 @@ def set_test_params(self):
     def skip_test_if_missing_module(self):
         self.skip_if_no_wallet()
 
-    def run_test(self):
-        filter_node = self.nodes[0].add_p2p_connection(FilterNode())
-
+    def test_size_limits(self, filter_node):
         self.log.info('Check that too large filter is rejected')
-        with self.nodes[0].assert_debug_log(['Misbehaving']):
-            filter_node.send_and_ping(msg_filterload(data=b'\xaa', nHashFuncs=MAX_BLOOM_HASH_FUNCS+1))
         with self.nodes[0].assert_debug_log(['Misbehaving']):
             filter_node.send_and_ping(msg_filterload(data=b'\xbb'*(MAX_BLOOM_FILTER_SIZE+1)))
 
+        self.log.info('Check that max size filter is accepted')
+        with self.nodes[0].assert_debug_log([], unexpected_msgs=['Misbehaving']):
+            filter_node.send_and_ping(msg_filterload(data=b'\xbb'*(MAX_BLOOM_FILTER_SIZE)))
+        filter_node.send_and_ping(msg_filterclear())
+
+        self.log.info('Check that filter with too many hash functions is rejected')
+        with self.nodes[0].assert_debug_log(['Misbehaving']):
+            filter_node.send_and_ping(msg_filterload(data=b'\xaa', nHashFuncs=MAX_BLOOM_HASH_FUNCS+1))
+
+        self.log.info('Check that filter with max hash functions is accepted')
+        with self.nodes[0].assert_debug_log([], unexpected_msgs=['Misbehaving']):
+            filter_node.send_and_ping(msg_filterload(data=b'\xaa', nHashFuncs=MAX_BLOOM_HASH_FUNCS, nTweak=0, nFlags=1))
```

Comment:
nit: explicitely setting `nTweak` and `nFlags` is not needed here

Reply:
Fixed, thanks!